specs/governance/alligator.md

```diff
@@ -0,0 +1,144 @@
+# Alligator
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Overview](#overview)
+- [Interface](#interface)
+  - [Core Functions](#core-functions)
+    - [`subdelegate`](#subdelegate)
+    - [`subdelegateBatched`](#subdelegatebatched)
+    - [`subdelegateBySig`](#subdelegatebysig)
+    - [`afterTokenTransfer`](#aftertokentransfer)
+  - [Getters](#getters)
+    - [`getSubdelegations`](#getsubdelegations)
+    - [`getCheckpoints`](#getcheckpoints)
+    - [`getVotingPower`](#getvotingpower)
+- [Storage](#storage)
+- [Backwards Compatibility](#backwards-compatibility)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+| Constant | Value                                        |
+|----------|----------------------------------------------|
+| Address  | `0x4200000000000000000000000000000000000043` |
+
+The `Alligator` contract implements subdelegations that can be used by token contracts inheriting from
+[`GovernanceToken`](gov-token.md). Subdelegations enable advanced delegation use cases, such as partial, time-constrained
+& block-based delegations, and relative & fixed allowances.
+
+The `Alligator` contract migrates the delegation state from a token contract to itself through a hook-based approach.
+Specifically, the token contract calls the `Alligator` contract's `afterTokenTransfer` function after a token
+transfer. This enables the `Alligator` contract to consume the hook and update its delegation and checkpoint mappings
+accordingly. If either address involved in the transfer (`_from_` or `_to`) has not been migrated to the `Alligator` contract,
+the contract copies the address' delegation and checkpoint data from the token contract to its own state.
+
+## Interface
+
+### Core Functions
+
+#### `subdelegate`
+
+Allows subdelegation of voting power to another address with specified subdelegation rules.
+
+```solidity
+subdelegate(address _to, SubdelegationRule _rule)
+```
+
+A subdelegation rule is an instance of the following struct:
+
+```solidity
+struct SubdelegationRule {
```

Comment:
It could be good to have a `## Types` section that includes the subdelegation rule + checkpoint with descriptions of each field in them outside of the code block

Reply:
agree, section added.